tests/integration/test_target_branch_protection.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Tests for the target_branch_protection function."""
+
+import typing
+from collections import namedtuple
+from uuid import uuid4
+
+import pytest
+from github import Consts
+from github.Branch import Branch
+
+from repo_policy_compliance import Result, target_branch_protection
+from repo_policy_compliance.exceptions import InputError
+from repo_policy_compliance.github_client import GITHUB_TOKEN_ENV_NAME
+
+
+def assert_substrings_in_string(substrings: typing.Iterable[str], string: str) -> None:
+    """Assert that a string contains substrings.
+
+    Args:
+        string: The string to check.
+        substrings: The sub strings that must be contained in the string.
+    """
+    for substring in substrings:
+        assert substring in string, f"{substring!r} not in {string!r}"  # nosec
+
+
+BranchProtectionParameters = namedtuple(
+    "BranchProtectionParameters",
+    [
+        "branch_protection_enabled",
+        "require_code_owner_reviews",
+        "dismiss_stale_reviews_enabled",
+        "bypass_pull_request_allowance_disabled",
+        "required_signatures_enabled",
+    ],
+)
+
+
+@pytest.mark.parametrize(
+    "github_branch, reason_string_array, branch_protection_parameters",
+    [
+        pytest.param(
+            f"not-protected/{uuid4()}",
+            ["not enabled"],
+            BranchProtectionParameters(
+                branch_protection_enabled=False,
+                require_code_owner_reviews=False,
+                dismiss_stale_reviews_enabled=False,
+                bypass_pull_request_allowance_disabled=False,
+                required_signatures_enabled=False,
+            ),
+            id="branch_protection disabled",
+        ),
+        pytest.param(
+            f"no-code-owner-review/{uuid4()}",
+            ["codeowner", "pull request", "review", "not required"],
+            BranchProtectionParameters(
+                branch_protection_enabled=True,
+                require_code_owner_reviews=False,
```

Comment:
Should we make everything `True` except for what we're checking to be wrong?

Reply:
Done